runtime/vm/createramclass.cpp

```diff
@@ -1577,6 +1577,42 @@ popLoadingOrLinkingStack(J9VMThread *vmThread, J9StackElement **stack, J9Pool *s
 	pool_removeElement(stackpool, topOfStack);
 }
 
+
+/**
+ * JEP 360: if super class/interface is sealed the inheriting subclass must be listed in the
+ * super's PermittedSubclasses attribute to be a legal subclass.
+ * @param superRomClass ROM class of super class or interface
+ * @param className name of subclass
+ * @param classNameLength length of subclass name
+ * @return TRUE if subclass can legally inherit the super, FALSE otherwise.
+ */
+static VMINLINE BOOLEAN
+isClassPermittedBySealedSuper(J9ROMClass *superRomClass, U_8* className, U_16 classNameLength)
+{
+	if (! J9ROMCLASS_IS_SEALED(superRomClass)) {
+		/* for non-sealed classes all subclasses are permitted (the final case is handled elsewhere). */
+		return TRUE;
+	}
+
+	BOOLEAN result = FALSE;
+	U_32 *permittedSubclassesCountPtr = getNumberOfPermittedSubclassesPtr(superRomClass);
+
+	/* find matching subclass name */
+	for (U_32 index = 0; index < *permittedSubclassesCountPtr; index++) {
+		J9UTF8* permittedSubclassNameUtf8 = permittedSubclassesNameAtIndex(permittedSubclassesCountPtr, index);
+		U_8 *permittedSubclassName = J9UTF8_DATA(permittedSubclassNameUtf8);
+		U_16 permittedSubclassLength = J9UTF8_LENGTH(permittedSubclassNameUtf8);
+
+		if ((permittedSubclassLength == classNameLength)
```

Comment:
Please use this macro instead (UTFs are not null terminated):

https://github.com/eclipse/openj9/blob/ee1f3d1ae159643f5cf8d3b3554224ee6282f082/runtime/oti/j9.h#L171